virtcontainers/vm.go

```diff
@@ -244,5 +233,8 @@ func (v *VM) assignSandbox(s *Sandbox) error {
 
 	s.hypervisor = v.hypervisor
 
+	// Use sanbox's agent since it points to the right communication channel.
+	v.agent = s.agent
```

Comment:
After `assignSandbox`, we should **never** use the VM pointer again. Why do you need the vm agent pointing to the right communication channel?

Reply:
the communication channel is used to initialize the VM (add cpus, memory, reseed RNG, etc), we are going to use the VM pointer again when the VM is initialized